Automattic/wp-calypso
# A4A: Update the Partner directories' labeling to use the brand URL.

Currently, in the expertise form, we have the directory list with these items:

- WordPress
- Jetpack
- Pressable
- WooCommerce

We have to display them with their URLs:

- WordPress.com
- Jetpack.com
- Pressable.com
- WooCommerce.com


| Before | After |
|--------|--------|
| <img width="765" alt="Screenshot 2024-06-21 at 12 14 22 AM" src="https://github.com/Automattic/wp-calypso/assets/56598660/8090490a-deae-4a7d-9c37-2c4244987f65"> | <img width="776" alt="Screenshot 2024-06-21 at 12 14 05 AM" src="https://github.com/Automattic/wp-calypso/assets/56598660/ff639470-2ac2-49dc-bea0-33cd54f5e3db"> | 

This PR addresses this.

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/697

## Proposed Changes

* Add a new record object to store available directories with the correct brand slug and URL. Update to use this new object.
* Remove no longer used constants and helper functions.


## Testing Instructions

* Use the A4A live link and go to the `/partner-directory/dashboard` page
* Confirm that any forms and elements referencing the directories use the brand URLs.

## Pre-merge Checklist
